documentation/modules/operators/ref-operator-cluster-feature-gate-releases.adoc

```diff
@@ -0,0 +1,77 @@
+// Module included in the following assemblies:
+//
+// assembly-using-the-cluster-operator.adoc
+
+[id='ref-operator-cluster-feature-gate-releases-{context}']
+= Feature gate releases
+
+[role="_abstract"]
+Feature gates have three stages of maturity:
+
+* Alpha — typically disabled by default
+* Beta — typically enabled by default
+* General Availability (GA) — typically always enabled
+
+Alpha stage features might be experimental or unstable, subject to change, or not sufficiently tested for production use.
+Beta stage features are well tested and their functionality is not likely to change.
+GA stage features are stable and should not change in the future.
+Alpha and beta stage features are removed if they do not prove to be useful.
+
+* The `ControlPlaneListener` feature gate moved to beta stage in Strimzi 0.27.0 and is expected to remain in the beta stage until Strimzi 0.31.
+* The `ServiceAccountPatching` feature gate moved to beta stage in Strimzi 0.27.0 and is expected to remain in the beta stage until Strimzi 0.30.
+* The `UseStrimziPodSets` feature gate is currently planned to move to the beta stage in Strimzi 0.30.0.
+
+NOTE: Feature gates might be removed when they reach GA. This means that the feature was incorporated into the Strimzi core features and can no longer be disabled.
+
+.Feature gates and the Strimzi versions when they moved to alpha, beta, or GA
+[cols="4*",options="header",stripes="none",separator=¦]
+|===
+
+¦Feature gate
+¦Alpha
+¦Beta
+¦GA
+
+¦`ControlPlaneListener`
+¦0.23.0
+¦0.27.0
+¦ -
+
+¦`ServiceAccountPatching`
+¦0.24.0
+¦0.27.0
+¦ -
+
+¦`UseStrimziPodSets`
+¦0.28.0
+¦ -
+¦ -
+
+|===
+
+== Disabling feature gates when upgrading Strimzi
+
+If a feature gate is enabled, you may need to disable it before upgrading or downgrading Strimzi.
+The following table shows which feature gates you need to disable when upgrading or downgrading Strimzi versions.
+
+.Disable feature gates when upgrading or downgrading
+[cols="3*",options="header",stripes="none",separator=¦]
+|===
+
+¦Disable Feature gate
+¦Upgrading from Strimzi version
+¦Downgrading to Strimzi version
+
+¦`ControlPlaneListener`
+¦0.22 and earlier
+¦0.22 and earlier
+
+¦`ServiceAccountPatching`
+¦-
+¦-
+
+¦`UseStrimziPodSets`
+¦-
+¦0.27 and earlier
+
+|===
```

Comment:
I don't think this is helpful too much => or rather not sufficient. This information needs to be in the part about the individual feature gates. Otherwise people will miss it. It probably also doesn't need the feature gate to which it doesn't apply in this table.

Reply:
Won't this table grow potentially? And other feature gates need this kind of information. Maybe it's good to have it in one place for easy maintenance and reference, with an _important_ note and link from the related feature gate descriptions section:

![image](https://user-images.githubusercontent.com/47596553/153889087-6c5163c7-e4ef-42c6-aee0-2fcc8762765f.png)

Otherwise, if we want this info with the feature gate descriptions, can we use attributes?

Reply:
Well, the question is for whom it is good to have it in one place. It is certainly easier for maintenance. But I'm not sure it is easier for the user.

My expectation is that users will be interested in particular feature gate. So the chapter about the specific feature gate should:
* explain whether it is enabled or disabled
* cover all important aspects such as upgrade or downgrade limitations
* possibly suggest when it might move to the next phase
* I do not think it necessarily needs to list when the phase changed last time

I think these things are too important to hide behind some generic link, they need to be placed directly.

When referring someone to it, you will again refer them to an exact feature gate because of some discussion about something. Not so some list of some upgrade issues. I do not think people will come and ask _Can you please point me to the list of FeatureGates with some special upgrade / downgrade issues?_. They will ask for some feature they need and you will point them to a specific Feature Gate related to it.

To be clear, I do not have problem to keep the table -> but the information needs to be in the other place as well.

Reply:
Thanks, @scholzj. I've adjusted the descriptions of each feature gate.